data/filter/FieldFilter.js

```diff
@@ -87,79 +87,72 @@ export class FieldFilter extends Filter {
                 };
+                break;
             case '>':
-                return r => {
-                    if (doNotFilter(r)) return true;
-                    const v = getVal(r);
-                    return !isNil(v) && v > value;
-                };
+                opTestFn = v => !isNil(v) && v > value;
+                break;
             case '>=':
-                return r => {
-                    if (doNotFilter(r)) return true;
-                    const v = getVal(r);
-                    return !isNil(v) && v >= value;
-                };
+                opTestFn = v => !isNil(v) && v >= value;
+                break;
             case '<':
-                return r => {
-                    if (doNotFilter(r)) return true;
-                    const v = getVal(r);
-                    return !isNil(v) && v < value;
-                };
+                opTestFn = v => !isNil(v) && v < value;
+                break;
             case '<=':
-                return r => {
-                    if (doNotFilter(r)) return true;
-                    const v = getVal(r);
-                    return !isNil(v) && v <= value;
-                };
+                opTestFn = v => !isNil(v) && v <= value;
+                break;
             case 'like':
                 regExps = value.map(v => new RegExp(escapeRegExp(v), 'i'));
-                return r => {
-                    if (doNotFilter(r)) return true;
-                    return regExps.some(re => re.test(getVal(r)));
-                };
+                opTestFn = v => regExps.some(re => re.test(v));
+                break;
             case 'not like':
                 regExps = value.map(v => new RegExp(escapeRegExp(v), 'i'));
-                return r => {
-                    if (doNotFilter(r)) return true;
-                    regExps.every(re => !re.test(getVal(r)));
-                };
+                opTestFn = v => regExps.every(re => !re.test(v));
+                break;
             case 'begins':
                 regExps = value.map(v => new RegExp('^' + escapeRegExp(v), 'i'));
-                return r => {
-                    if (doNotFilter(r)) return true;
-                    regExps.some(re => re.test(getVal(r)));
-                };
+                opTestFn = v => regExps.some(re => re.test(v));
+                break;
             case 'ends':
                 regExps = value.map(v => new RegExp(escapeRegExp(v) + '$', 'i'));
-                return r => {
-                    if (doNotFilter(r)) return true;
-                    regExps.some(re => re.test(getVal(r)));
-                };
+                opTestFn = v => regExps.some(re => re.test(v));
+                break;
             default:
                 throw XH.exception(`Unknown operator: ${op}`);
         }
+
+        if (!store) return r => opTestFn(r[field]);
+
+        // Pass/fail depends on the filter value mode of the store
+        // Uncommitted records always pass the filter
```

Comment:
Assuming we keep the "adds always pass" behavior, should we move that check up and short-circuit the creation of the filter functions entirely?

Reply:
Yeah I wasn't sure about this, I considered adding an additional boolean to Store to control this behavior but in the end after playing around with it in the inline editing example, I just found it clearly undesirable to ever click an 'Add Row' button and have nothing at all appear to happen in the grid, regardless of the filter value mode selected, and I couldn't think of a use-case where you are dealing with uncommitted records and would want them to be filtered out since I think that implies that a record was added interactively by the user

Reply:
Also I wonder if there are use-cases where an app would want to have modified records always pass the filter also?